tests/test_basic_functionality.py

```diff
@@ -458,6 +458,18 @@ def test_to_dict_yaml(test_spec: Path) -> None:
     assert yaml_config == lct.to_dict()
 
 
+def test_getter() -> None:
+    getter_dict: NestedDict = {
+        "outer_layer_1": "test_value",
+        "outer_layer_2": {"inner_layer": "test_value2"},
+    }
+    lct = LayeredConfigTree(getter_dict)
+
+    assert lct.get("outer_layer_1") == "test_value"
+    assert lct.get("outer_layer_2").to_dict() == getter_dict["outer_layer_2"]  # type: ignore [union-attr]
```

Comment:
nit: slightly more accurate (I think?) to do `assert lct.get("outer_layer_2") == NestedDict(getter_dict["outer_layer_2"]_`

Reply:
You'll remember our discussion - NestedDict is a type rather than a class

Reply:
In the same vein as Steve's comment, what about this:
```python
assert lct.get("outer_layer_2") == LayeredConfigTree(getter_dict["outer_layer_2"])
```

Reply:
I tried this originally but LayeredConfigTree doesn't have an equality operator. Do we want to add one? Converting to a dictionary does make us lose information about layers

Reply:
Use asserts to check type